packages/astro/src/content/loaders/glob.ts

```diff
@@ -124,7 +129,10 @@ export function glob(globOptions: GlobOptions): Loader {
 					filePath,
 				});
 
-				if (entryType.getRenderFunction) {
+				if (entryType.extensions.includes('.mdx')) {
+					store.addModuleImport(relativePath);
```

Comment:
Could this be called internally when `set` is called with `deferredRender` set to true?